#! /bin/sh /usr/share/dpatch/dpatch-run
## 19-fix-taoidl-fd-leak.dpatch by Thomas Girard <thomas.g.girard@free.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix a file descriptor leak in taoidl.
## DP: See ChangeLog entry
## DP:  "Tue Jan 11 14:49:40 2005  Jeff Parsons <j.parsons@vanderbilt.edu>"

@DPATCH@
--- ACE_wrappers/TAO/TAO_IDL/driver/drv_preproc.cpp.org	2005-02-14 15:09:19.000000000 +0100
+++ ACE_wrappers/TAO/TAO_IDL/driver/drv_preproc.cpp	2005-02-14 15:12:03.000000000 +0100
@@ -681,6 +681,8 @@
   char * t_file  = tmp_cpp_file;
   char * t_ifile = tmp_cpp_ifile;
 
+  ACE_OS::close (tf_fd);
+
 #endif  /* ACE_LACKS_MKSTEMP */
 
   // Rename temporary files so that they have extensions accepted
